src/DotNetCore.CAP.AzureServiceBus/AzureServiceBusConsumerClient.cs

```diff
@@ -78,45 +81,51 @@ public void Listening(TimeSpan timeout, CancellationToken cancellationToken)
         {
             ConnectAsync().GetAwaiter().GetResult();
 
-            if (_asbOptions.EnableSessions)
-            {
-                _consumerClient!.RegisterSessionHandler(OnConsumerReceivedWithSession,
-                    new SessionHandlerOptions(OnExceptionReceived)
-                    {
-                        AutoComplete = false,
-                        MaxAutoRenewDuration = TimeSpan.FromSeconds(30)
-                    });
-            }
-            else
-            {
-                _consumerClient!.RegisterMessageHandler(OnConsumerReceived,
-                    new MessageHandlerOptions(OnExceptionReceived)
-                    {
-                        AutoComplete = false,
-                        MaxConcurrentCalls = 10,
-                        MaxAutoRenewDuration = TimeSpan.FromSeconds(30)
-                    });
-            }
+            _serviceBusProcessor.ProcessMessageAsync += _serviceBusProcessor_ProcessMessageAsync;
+            _serviceBusProcessor.ProcessErrorAsync += _serviceBusProcessor_ProcessErrorAsync;
+
+            _serviceBusProcessor.StartProcessingAsync().GetAwaiter().GetResult();
+
+        }
+
+        private Task _serviceBusProcessor_ProcessErrorAsync(ProcessErrorEventArgs args)
+        {
+            
+            var exceptionMessage =
+                $"- Identifier: {args.Identifier}" + Environment.NewLine +
+                $"- Entity Path: {args.EntityPath}" + Environment.NewLine +
+                $"- Executing ErrorSource: {args.ErrorSource}" + Environment.NewLine +
+                $"- Exception: {args.Exception}";
 
-            while (true)
+            var logArgs = new LogMessageEventArgs
             {
-                cancellationToken.ThrowIfCancellationRequested();
-                cancellationToken.WaitHandle.WaitOne(timeout);
-            }
-            // ReSharper disable once FunctionNeverReturns
+                LogType = MqLogType.ExceptionReceived,
+                Reason = exceptionMessage
+            };
+
+            OnLogCallback!(logArgs);
+
+            return Task.CompletedTask;
+        }
+
+        private async Task _serviceBusProcessor_ProcessMessageAsync(ProcessMessageEventArgs arg)
+        {
+            var context = ConvertMessage(arg.Message);
+            
+            await OnMessageCallback!(context, new AzureServiceBusConsumerCommitInput(arg.Message.LockToken));
         }
 
         public void Commit(object? sender)
         {
-            var commitInput = (AzureServiceBusConsumerCommitInput) sender!;
-            if (_asbOptions.EnableSessions)
-            {
-                commitInput.Session?.CompleteAsync(commitInput.LockToken);
-            }
-            else
-            {
-                _consumerClient!.CompleteAsync(commitInput.LockToken);
-            }
+            //var commitInput = (AzureServiceBusConsumerCommitInput) sender!;
```

Comment:
Hey @giorgilekveishvili-meama. Could you clarify why is this code here commented? If the `AutoCompleteMessages` property is set to false, the message must be manually completed. Check [this section on the Migration Guide from Microsoft](https://learn.microsoft.com/en-us/azure/service-bus-messaging/message-transfers-locks-settlement#settling-receive-operations).

Thank you!

Reply:
Just adding for registration: the `Commit` method is called [here](https://github.com/dotnetcore/CAP/blob/master/src/DotNetCore.CAP/Internal/IConsumerRegister.Default.cs) when storing a message in the Inbox.